fast64_internal/oot/c_writer/oot_level_c.py

```diff
@@ -382,33 +407,26 @@ def ootRoomCommandsToC(room, headerIndex):
     return data
 
 
-def ootAlternateRoomMainToC(scene, room):
+def ootAlternateRoomMainToC(scene: OOTScene, room: OOTRoom):
     altHeader = CData()
     altData = CData()
 
     altHeader.header = "extern SCmdBase* " + room.alternateHeadersName() + "[];\n"
     altHeader.source = "SCmdBase* " + room.alternateHeadersName() + "[] = {\n"
 
-    if room.childNightHeader is not None:
-        altHeader.source += "\t" + room.roomName() + "_header" + format(1, "02") + ",\n"
-        altData.append(ootRoomMainToC(scene, room.childNightHeader, 1))
-    else:
-        altHeader.source += "\t0,\n"
-
-    if room.adultDayHeader is not None:
-        altHeader.source += "\t" + room.roomName() + "_header" + format(2, "02") + ",\n"
-        altData.append(ootRoomMainToC(scene, room.adultDayHeader, 2))
-    else:
-        altHeader.source += "\t0,\n"
-
-    if room.adultNightHeader is not None:
-        altHeader.source += "\t" + room.roomName() + "_header" + format(3, "02") + ",\n"
-        altData.append(ootRoomMainToC(scene, room.adultNightHeader, 3))
-    else:
-        altHeader.source += "\t0,\n"
+    for headerIndex, headerName in enumerate(["childNightHeader", "adultDayHeader", "adultNightHeader"], 1):
+        # equivalent to ``room.childNightHeader`` if ``headerName`` is "childNightHeader"
+        curHeader = getattr(room, headerName)
+        if curHeader is not None:
+            altHeader.source += indent + f"{room.roomName()}_header{headerIndex:02},\n"
+            altData.source += ootGetHeaderDefines(curHeader, headerIndex)
+            altData.append(ootRoomMainToC(scene, curHeader, headerIndex))
```

Comment:
why += `ootGetHeaderDefines()` here instead of in `ootRoomMainToC` ?

Couldn't `+= ootGetHeaderDefines(room, 0)` in `ootRoomMainToC` have `headerIndex` instead of 0 ?